cfgov/form_explainer/models/blocks.py

```diff
@@ -0,0 +1,75 @@
+from django.core.exceptions import ValidationError
+from django.forms.utils import ErrorList
+
+from wagtail.wagtailcore import blocks
+from wagtail.wagtailimages import blocks as images_blocks
+
+
+class ImageMapCoordinates(blocks.StructBlock):
+    left = blocks.FloatBlock(required=True, min_value=0, max_value=100)
+    top = blocks.FloatBlock(required=True, min_value=0, max_value=100)
+    width = blocks.FloatBlock(required=True, min_value=0, max_value=100)
+    height = blocks.FloatBlock(required=True, min_value=0, max_value=100)
+
+    def clean(self, value):
+        cleaned = super(ImageMapCoordinates, self).clean(value)
+        errors = {}
+        if cleaned.get('left') + cleaned.get('width') > 100:
+            errors['left'] = errors['width'] = ErrorList([
+                'Sum of left and width values should not exceed 100.'
+            ])
+        if cleaned.get('top') + cleaned.get('height') > 100:
+            errors['top'] = errors['height'] = ErrorList([
+                'Sum of top and height values should not exceed 100.'
+            ])
+        if errors:
+            raise ValidationError(
+                'Validation error in ImageMapCoordinates',
+                params=errors
+            )
+        return cleaned
+
+
+class ExplainerNote(blocks.StructBlock):
+    coordinates = ImageMapCoordinates(
+        form_classname='coordinates',
+        label='Note image map coordinates',
+        help_text='Enter percentage values to define the area '
+                  'that will be highlighted on the image for this note.')
+    heading = blocks.CharBlock(required=True, label='Note heading')
+    body = blocks.RichTextBlock(
+        required=True,
+        features=[
+            'bold', 'italic', 'link', 'document-link'
+        ],
+        label='Note text')
+
+
+class ExplainerCategory(blocks.StructBlock):
+    title = blocks.CharBlock(
+        required=False,
+        label='Category title',
+        help_text='Optional. Leave blank if there is only '
+                  'one type of note for this image.'
+    )
+    notes = blocks.ListBlock(ExplainerNote(
```

Comment:
Minor thing, but if you create a page, and then add an Explainer block, and hit preview to see what it looks like, you get validation errors if you haven't entered any notes. To avoid this, you could modify this definition to be `blocks.ListBlock(..., default[])` so it defaults to the closed state:

![image](https://user-images.githubusercontent.com/654645/75713289-8f2b0600-5c97-11ea-9d8d-fb7321c36066.png)

Reply:
Updated in 163449e6b38bc020ec9abfa0fdc5ebea79095c8d